sdk/data/azcosmos/cosmos_transactional_batch_response.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+// TransactionalBatchResponse contains response from a transactional batch operation.
+type TransactionalBatchResponse struct {
+	Response
+	// SessionToken contains the value from the session token header to be used on session consistency.
+	SessionToken string
+	// OperationResults contains the individual batch operation results.
+	// The order of the results is the same as the order of the operations in the batch.
+	OperationResults []TransactionalBatchResult
+	// Success indicates if the transaction was successfully committed.
+	// If false, one of the operations in the batch failed.
+	// Inspect the OperationResults, any operation with status code http.StatusFailedDependency is a dependency failure.
+	// The cause of the batch failure is the first operation with status code different from http.StatusFailedDependency.
+	Success bool
+}
+
+func newTransactionalBatchResponse(resp *http.Response) (TransactionalBatchResponse, error) {
+	response := TransactionalBatchResponse{
+		Response: newResponse(resp),
+	}
+
+	response.SessionToken = resp.Header.Get(cosmosHeaderSessionToken)
+
+	response.Success = resp.StatusCode != http.StatusMultiStatus
```

Comment:
Are 4xx and 5xx status codes handled elsewhere?

Reply:
Yes, those would be returned as errors, not a response. MultiStatus is a 207 (that's the status code the service picked) so it's not a failure, but a valid response, which the user needs to read and understand.

Reply:
The contract that was defined service side is, for scenarios where one of the operations generates a rollback, it returns a 207 (instead of 200) and the individual operation result should be inspected (same as it needs to be done on other language SDKs). If the service returns a 4xx or 5xx, then that is a failure and the call returns the `err` to the caller.